src/controllers/authController.js

```diff
@@ -1,94 +1,57 @@
 const User = require('../models/User');
-const jwt = require('jsonwebtoken');
 const argon2 = require('argon2');
-
-// TODO: SET up CUSTOM ERRORs later
-
-// create token
-const createJWT = (user) => {
-  const payload = {
-    name: user.name,
-    userId: user._id,
-    role: user.role,
-  };
-  const token = jwt.sign(payload, process.env.JWT_SECRET, {
-    expiresIn: process.env.JWT_LIFETIME,
-  });
-  return token;
-};
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+const { attachCookiesToResponse, createTokenUser } = require('../utils');
 
 const register = async (req, res) => {
-  const { name, email, password } = req.body;
-
-  const emailAlreadyExists = await User.findOne({ email });
+  const { email, name, password } = req.body; // Extract data from the request
```

Comment:
This code should also check if `email`, `name`, and `password` were passed in/not empty from the request, just like you've done on line35 for the login handler. Otherwise, we may end up returning weird errors if the user only remembers to enter for example only their name but not their email/password

Reply:
<img width="503" alt="Снимок экрана 2023-08-09 174821" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/fd84c381-3702-44d9-851e-22dca91008f4">